infra/test-core2js.rkt

```diff
@@ -0,0 +1,109 @@
+#lang racket
+
+(require "../tools/common.rkt" "../tools/core2js.rkt" "../tools/fpcore.rkt")
+
+(define tests-to-run (make-parameter 10))
+(define test-file (make-parameter "/tmp/test.js"))
+(define fuel (make-parameter 1000))
+
+(define ((eval-fuel evaltor fuel [default #f]) expr ctx)
+  (let/ec k
+    (let eval ([expr expr] [ctx ctx] [fuel fuel])
+      (if (<= fuel 0)
+        (k default)
+        ((eval-expr* evaltor (λ (expr ctx) (eval expr ctx (- fuel 1)))) expr ctx)))))
+
+;; TODO: add support for multiple types
+(define (random-exp k)
+  "Like (random (expt 2 k)), but k is allowed to be arbitrarily large"
+  (if (< k 31) ; Racket generates random numbers in the range [0, 2^32-2]; I think it's a bug
+      (random (expt 2 k))
+      (let ([head (* (expt 2 31) (random-exp (- k 31)))])
+        (+ head (random (expt 2 31))))))
+
+(define (sample-double)
+  (floating-point-bytes->real (integer->integer-bytes (random-exp 64) 8 #f)))
+
+(define (compile->js prog test-file)
+  (call-with-output-file test-file #:exists 'replace
+    (λ (p)
+       (define N (length (second prog)))
+       (fprintf p "math = require('mathjs')\n\n~a\n\n" (compile-program prog #:name "f"))
+       (fprintf p "console.log(f(~a));\n"
+                (string-join (map (curry format "~a(process.argv[~a])" 'parseFloat) 
+                                  (map ((curry +) 2) (range N))) ", "))))
+  test-file)
+
+(define (run<-js file-name ctx)
+  (define out
+    (with-output-to-string
+      (λ ()
+         (define file-command (format "node ~a" file-name))
+         (system (string-join (cons file-command (map (compose ~a real->double-flonum)
+                                                      (dict-values ctx)))
+                              " ")))))
+  (define out*
+    (match out
+      ["NaN\n" "+nan.0"]
+      ["Infinity\n" "+inf.0"]
+      ["-Infinity\n" "-inf.0"]
+      [x (substring x 0 (sub1 (string-length x)))]))
+  ;; javascript can return imaginary numbers which the reference implementation
+  ;; doesn't have (returns NaN)
```

Comment:
I think you should note in this comment that that's because you're using mathjs operations. Normal Javascript `Math` operations don't return complex numbers.